components/search/src/types.rs

```diff
@@ -0,0 +1,179 @@
+    }
+}
+
+/// The user's environment that is used for filtering the search configuration.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchUserEnvironment {
+    /// The current locale of the application that the user is using.
+    pub locale: String,
+
+    /// The home region that the user is currently identified as being within.
+    /// On desktop & android there is a 14 day lag after detecting a region
+    /// change before the home region changes. TBD: iOS?
+    pub region: String,
+
+    /// The update channel of the user's build.
+    pub update_channel: SearchUpdateChannel,
+
+    /// The distribution id for the user's build.
+    pub distribution_id: String,
+
+    /// The search related experiment id that the user is included within. On
+    /// desktop this is the `searchConfiguration.experiment` variable.
+    pub experiment: String,
+
+    /// The application name that the user is using.
+    pub app_name: SearchApplicationName,
+
+    /// The application version that the user is using.
+    pub version: String,
+}
+
+/// Parameter definitions for search engine URLs. The name property is always
+/// specified, along with one of value, experiment_config or search_access_point.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchUrlParam {
+    /// The name of the parameter in the url.
+    pub name: String,
+    /// The parameter value, this may be a static value, or additionally contain
+    /// a parameter replacement, e.g. `{inputEncoding}`. For the partner code
+    /// parameter, this field should be `{partnerCode}`.
+    pub value: Option<String>,
+    /// The value for the parameter will be derived from the equivalent experiment
+    /// configuration value.
+    /// Only desktop uses this currently.
+    pub experiment_config: Option<String>,
+}
+
+/// Defines an individual search engine URL.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchEngineUrl {
+    /// The PrePath and FilePath of the URL. May include variables for engines
+    /// which have a variable FilePath, e.g. `{searchTerm}` for when a search
+    /// term is within the path of the url.
+    pub base: String,
+
+    /// The HTTP method to use to send the request (`GET` or `POST`).
+    /// If not specified, defaults to GET.
+    pub method: Option<String>,
+
+    /// The parameters for this URL.
+    pub params: Option<Vec<SearchUrlParam>>,
+
+    /// The name of the query parameter for the search term. Automatically
+    /// appended to the end of the query. This may be skipped if `{searchTerm}`
+    /// is included in the base.
+    pub search_term_param_name: Option<String>,
+}
+
+/// The URLs associated with the search engine.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchEngineUrls {
+    /// The URL to use for searches.
+    pub search: SearchEngineUrl,
+
+    /// The URL to use for suggestions.
+    pub suggestions: Option<SearchEngineUrl>,
+
+    /// The URL to use for trending suggestions.
+    pub trending: Option<SearchEngineUrl>,
```

Comment:
Just curious, do we have this for Android?

Reply:
Not currently, this is a relatively new feature on desktop that we introduced earlier in the year.

Reply:
Oh! This would actually be pretty helpful, as one of the open questions I have about the Trending Searches request that we've been asked to roadmap is "what API is providing trending search suggestions for us?" Sounds like this might be a simple opportunity to carry that functionality over from Desktop into Android without having to re-think it 😄 

I'm going to link this comment in that ticket, just for transparency! No clue if it'll wind up useful.